app/presenters/checkout_presenter.rb

```diff
@@ -47,9 +47,10 @@ def checkout_props(params:, browser_guid:)
 
   def checkout_product(product, cart_item, params, include_cross_sells: true)
     return unless product.present?
-    upsell_variants = product.upsell&.upsell_variants&.alive
+    upsell_variants = product.upsell_variants.alive.includes(:selected_variant, :offered_variant)
     bundle_products = product.bundle_products.in_order.includes(:product, :variant).alive.load
     accepted_offer = params[:accepted_offer_id] ? Upsell.alive.where(product:).find_by_external_id!(params[:accepted_offer_id]) : nil
+    option_id = accepted_offer&.variant&.external_id || cart_item[:option]&.fetch(:id)
```

Comment:
If the accepted offer has a defined variant, we should select it instead of the default, first available option